### PR TITLE
better docs for Chef::Knife::Bootstrap#validate_options!

### PR DESCRIPTION
**Pull request by bankair**
This PR add some documentation for method
Chef::Knife::Bootstrap#validate_options!.

- [x] All tests pass
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: lib/chef/knife/bootstrap.rb [hunk 1]
```diff
@@ -409,7 +409,14 @@ def validate_name_args!
...
       end
 
-      # make sure policyfile values are set correctly
+      # Ensure options are valid by checking policyfile values.
+      #
+      # The method call will cause the program to exit(1) if:
+      #   * Only one of --policy-name and --policy-group is specified
+      #   * Policyfile options are set and --run-list is set as well
+      #
+      # === Returns
```
Comment (tas50) [inline lib/chef/knife/bootstrap.rb]: Can you change this  over to YARD format. I realize we're not YARD everywhere, but we're trying to slowly document the world. https://gist.github.com/chetan/1827484

Comment (bankair) [reply lib/chef/knife/bootstrap.rb]: Yeah, sure. Is there a specific for exit(1) in yard?

Comment (tas50) [reply lib/chef/knife/bootstrap.rb]: I think this is just @return [TrueClass] If options are valid.

**Comment by bankair** [comment]
@tas50 : This way?

**Pull request merged**